server/src/modules/authentication/authentication.service.ts

```diff
@@ -0,0 +1,196 @@
+import { AuthenticationSettings } from './entities/authentication_settings.entities';
+import * as bcrypt from 'bcrypt';
+import * as crypto from 'crypto';
+import * as jwt from 'jsonwebtoken';
+import { randomUUID } from 'crypto';
+
+@Injectable()
+export class AuthenticationService implements OnModuleInit {
+  constructor(
+    @InjectRepository(AuthenticationSettings)
+    private readonly authenticationSettingsRepository: Repository<AuthenticationSettings>,
+  ) {}
+
+  generateApiKey(): string {
+    return Buffer.from(`Maintainerr${Date.now()}${randomUUID()}`).toString(
+      'base64',
+    );
+  }
+  generateJwt_secret(): string {
+    return crypto.randomBytes(32).toString('hex');
+  }
+
+  async onModuleInit() {
+    let settings = await this.authenticationSettingsRepository.findOne({
+      where: {},
+    });
+
+    if (!settings) {
+      settings = this.authenticationSettingsRepository.create({
+        authEnabled: false, // ✅ Default authentication disabled
+        username: null,
+        passwordHash: null,
+        jwt_secret: this.generateJwt_secret(), // ✅ Store JWT secret
+        apiKey: this.generateApiKey(), // ✅ Generate API key
+      });
+
+      await this.authenticationSettingsRepository.save(settings);
+    }
+  }
+
+  async getJwtSecret(): Promise<string> {
+    const settings = await this.authenticationSettingsRepository.findOne({
+      where: {},
+    });
+
+    if (!settings || !settings.jwt_secret) {
+      throw new Error('JWT secret is missing from the database!');
+    }
+
+    return settings.jwt_secret;
+  }
+
+  async login(
+    username: string,
+    password: string,
+    res: Response,
+  ): Promise<{ success: boolean; message: string }> {
+    const settings = await this.getAuthenticationSettings();
+    const secret = await this.getJwtSecret();
+
+    if (!settings.authEnabled) {
+      return { success: false, message: 'Authentication is disabled' };
+    }
+
+    if (!settings.username || !settings.passwordHash) {
+      throw new UnauthorizedException('Invalid credentials');
+    }
+
+    const isValidUser = username === settings.username;
+    const isValidPassword = await bcrypt.compare(
+      password,
+      settings.passwordHash,
+    );
+
+    if (!isValidUser || !isValidPassword) {
+      throw new UnauthorizedException('Invalid credentials');
+    }
+
+    // Generating AccessToken
+    const accessToken = jwt.sign({ username }, secret, { expiresIn: '1h' });
+
+    // Generating RefreshToken
+    const refreshToken = jwt.sign({ username }, secret, { expiresIn: '30d' });
+
+    res.cookie('sessionToken', accessToken, {
+      httpOnly: true,
+      secure: process.env.NODE_ENV === 'production',
+      sameSite: 'lax',
+      path: '/',
+      maxAge: 60 * 60, // 1h in seconds
+    });
+
+    res.cookie('refreshToken', refreshToken, {
+      httpOnly: true,
+      secure: process.env.NODE_ENV === 'production',
+      sameSite: 'lax',
+      path: '/',
```

Comment:
The path should be set to the refresh token endpoint, taking into account any subfolder config. Currently you're sending the refresh token along with the access token which mostly defeats the point of the refresh token. This change will break the middleware refresh you do so you'll need to move the refresh to the client as I suggested.